ARMmbed/mbl-tools
# Add support for creating binary release archives

For IOTMBL-1535: Make sure output of OE build contains necessary items
for evaluation images

build.sh:
* Add code to create new artifacts required for binary releases:
  - a README file;
  - license manifests;
  - a build info file;
  - the binary release archive itself.
* When archiving source, set COPYLEFT_LICENSE_INCLUDE = "*" in BitBake's
  local.conf - otherwise the default value is used, which results in
  only archiving the source for recipes with GPL flavoured licenses.
* Add command line parameters for:
  - building binary releases;
  - passing information about the version of mbl-tools being used;
  - passing information about how the parent script (probably run-me.sh)
    was invoked.
  The info being passed from the parent script is used to populate the
  build info file that is added to binary release archives.

run-me.sh:
* Add a command line parameter to pass information about the version of
  mbl-tools being used (to be put into the build info file).
* Pass run-me.sh's command line to build.sh (to be put into the build
  info file).

binary_release_README:
* Add a template README file for binary releases.

Dockerfile:
* Put the template README file somewhere where build.sh can use it to
  create an actual README for a binary release.

**Jenkins builds**
* http://jenkins.mbed-linux.arm.com/view/jh/job/jh-test/54/ (failed for imx8; standard build with this branch of mbl-tools -- not using `--binary-release` flag).
* http://jenkins.mbed-linux.arm.com/view/jh/job/jh-test/55/ (succeeded; after changes; standard build with this branch of mbl-tools -- not using `--binary-release` flag).
* http://jenkins.mbed-linux.arm.com/view/jh/job/jh-test/57/ (in progress; after more changes; standard build with this branch of mbl-tools -- not using `--binary-release` flag).

**Testing done**
Before changes for review comments:
* Build mbl-image-development using the new "--binary-release" flag and check that the contents of the resulting archive look reasonable.
* Build mbl-image-production using `--binary-release` and `--no-compress` and check that the contents of the resulting archive look reasonable.
* Check that the images inside the mbl-image-production binary release archive for warp7 actually boot.
* Build without using `--binary-release` (for rpi3, warp7, pico).

After changes for review comments:
* Build mbl-image-development using --binary-release" flag and check that the contents of the resulting archive look reasonable. Not specifying "--mbl-tools-version" and letting run-me.sh work it out.
* Build mbl-image-development using "--binary-release" flag and check that the contents of the resulting archive look reasonable. Specifying "--mbl-tools-version".
* Build mbl-image-development using "--binary-release" flag and check that the contents of the resulting archive look reasonable. Not specifying "--mbl-tools-version" and running from a place where run-me.sh can't work out the mbl-tools version.
